### PR TITLE
Display confirmation modal before updating library

### PR DESCRIPTION
**Pull request by maddiedierker**
Before, when you clicked the "update" button (shown below) in the "Manage Libraries" modal, it would automatically update the library and reload the page.
Screen Shot 2020-03-31 at 3 22 26 PM

Now, we display a confirmation modal before updating the library, which allows the user to see the code in the updated library version. [Video of new behavior](https://drive.google.com/file/d/1TlfBGdR8-ct5pEB-QzxaJ5n_r0Kk-9aJ/view?usp=sharing)

I also made some styling changes to avoid clipping really long library names. Examples:

1. A library name that takes up more than one line:
Screen Shot 2020-04-01 at 4 57 44 PM

2. A title that needs to wrap onto two lines:
Screen Shot 2020-04-01 at 4 57 16 PM

## Links

- [STAR-1010](https://codedotorg.atlassian.net/browse/STAR-1010)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx [hunk 1]
```diff
@@ -286,34 +298,89 @@ export class LibraryManagerDialog extends React.Component {
...
+      return;
+    }
+
+    const onClose = () =>
+      this.setState({
+        displayLibrary: null,
+        displayLibraryMode: DisplayLibraryMode.NONE
+      });
+
+    if (displayLibraryMode === DisplayLibraryMode.VIEW) {
```
Comment (ajpal) [inline apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx]: nit: slight preference for a switch statement here, which would make it explicit that you cover each case of the enum, and would make it easier to catch if we add another mode. Not blocking though!

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx [hunk 2]
```diff
@@ -286,34 +298,89 @@ export class LibraryManagerDialog extends React.Component {
...
-      error
-    } = this.state;
+    const {importLibraryId, displayLibrary, isLoading, error} = this.state;
+
+    if (!isOpen) {
+      return null;
+    }
+
     return (
       <div>
```
Comment (ajpal) [inline apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx:377]: is this wrapping div now unnecessary?

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx [hunk 3]
```diff
@@ -286,34 +298,89 @@ export class LibraryManagerDialog extends React.Component {
...
         <BaseDialog
-          isOpen={isOpen}
+          isOpen={true}
```
Comment (ajpal) [inline apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx]: nit: Can just write as `isOpen`

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx (hunk 2 above)
Comment (maddiedierker) [reply apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx:377]: no, there are still 2 elements within this div (`<BaseDialog/>` and `<LibraryViewCode/>`)

Comment (ajpal) [reply apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx:377]: ah nevermind, sorry!

**Review by ajpal — APPROVED** [review]

File: apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx (hunk 1 above)
Comment (maddiedierker) [reply apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx]: that makes sense to me -- i'll add `return null;` as the default case for now

**Pull request merged**